### DISCUSSION TITLE
Repeating textures

### DISCUSSION THREAD

**Discussion by arcanis** [raysan5/raylib#4267] [Q&A] [answered]
Is it expected that `SetTextureWrap(_, TEXTURE_WRAP_REPEAT)` has seemingly no effect on `DrawTexturePro`? Is there a way to make it work without manually looping over each individual repetition?

**Comment by Peter0x44** [comment, marked as answer]
@arcanis The texture repeat (GL_REPEAT) is the default, afaik. However, not all OpenGL versions support it, specifically I know of OpenGL ES2. For web, to have this behavior, you need to build raylib for OpenGL ES3 and link with `-sFULL_ES3`.

If you want proof it works:
https://tradam.itch.io/raylib-drawtexturepro-interactive-demo

I don't know if web is relevant, but you never specified platform or OpenGL version...
Need more info to tell you anything concrete.

**Reply by arcanis** [reply]
Ahh I didn't understand that I needed the source to be > than the actual image size for it to work - I thought increasing the destination was enough. Nice little tool you have here, thanks!
